src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -98,14 +115,18 @@ public function normalize($object, $format = null, array $context = [])
         $attributesMetadata = $this->classMetadataFactory ? $this->classMetadataFactory->getMetadataFor($class)->getAttributesMetadata() : null;
         $maxDepthHandler = $context[self::MAX_DEPTH_HANDLER] ?? $this->defaultContext[self::MAX_DEPTH_HANDLER] ?? $this->maxDepthHandler;
 
+        if (null !== $maxDepthHandler && !\is_callable($maxDepthHandler)) {
+            throw new InvalidArgumentException(sprintf('The %s given in the context is not callable.', self::MAX_DEPTH_HANDLER));
+        }
+
         foreach ($attributes as $attribute) {
             $maxDepthReached = false;
             if (null !== $attributesMetadata && ($maxDepthReached = $this->isMaxDepthReached($attributesMetadata, $class, $attribute, $context)) && !$maxDepthHandler) {
                 continue;
             }
 
             $attributeValue = $this->getAttributeValue($object, $attribute, $format, $context);
-            if ($maxDepthReached) {
+            if ($maxDepthReached && null !== $maxDepthHandler) {
```

Comment:
how did this work before? would it error if there was no maxDepthHandler?

Reply:
Thanks